katran/lib/KatranLb.h

```diff
@@ -224,6 +224,18 @@ class KatranLb {
    */
   bool delRealForVip(const NewReal& real, const VipKey& vip);
 
+  /**
+   * @param ModifyAction action. either ADD or DEL
+   * @param NewReal real to be marked/unmarked as local
+   * @param VipKey vip from which we want to modify specified real
+   * @return true on success
+   *
+   * helper function to mark specified real from vip as local
+   * returns true if real does exist for specified vip and modified successfully
+   * could throw if real's address cant be parsed to v4 or v6
+   */
+  bool modifyLocalMarkForReal(const ModifyAction action, const NewReal& real, const VipKey& vip);
```

Comment:
imo this should be generic helper to modify flags for real (e.g. so it would be easier to add any new flag in future). user of the library could add anything on top of that.
we already have modifyVip (line 187 in this file) which allows to set/unset flags for vip. as well as addVip (line 160) which allows to specify custom flags during vip creation. 

so i think what we need to have is the same as modifyVip. e.g. something like 
modifyReal(const std::string& real, uint8_t flags, bool set = true)
that helper should:
1. check if real exists in reals_. 
 1.1 if not - do nothing. 
 1.2 if it is exists -> modify it flags (RealMeta should be updated to include flags as well) and run updateRealsMap (should be modified to include flags as well)

should be way simpler that what you have here and lots of things could be reused.


so the user of the library to achive local deliver optimization would just

1. addVip w/ local flag
2. addReals for that vip
3. run modifyReal w/ local address and "local" flag

and it could enable/disable this local delivery optimization in runtime either for specific vip (by running modifyVip w/ "set" equal to false) or for all vips by runnign modifyReal w/ "set" equal to false.